### PR TITLE
WINDUP-685: Inappropriate output directory selection in some cases

### PR DESCRIPTION
**Pull request by jsight**
https://issues.jboss.org/browse/WINDUP-685

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/901/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/270/
Test PASSed.

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/decompiler/DecompilerUtil.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     */
+    static File getOutputDirectoryForClass(GraphContext context, JavaClassFileModel fileModel)
+    {
+        final File result;
+        WindupConfigurationModel configuration = WindupConfigurationService.getConfigurationModel(context);
+        String inputPath = configuration.getInputPath().getFilePath();
+        if (fileModel.getFilePath().startsWith(inputPath))
+        {
+            String outputPath = configuration.getOutputPath().getFilePath();
+            result = Paths.get(outputPath).resolve("classes").toFile();
```
Comment (mbriskar) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/decompiler/DecompilerUtil.java]: isn't that needed to preserve the package in the file path?

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/decompiler/DecompilerUtil.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ */
+public class DecompilerUtil
+{
+    /**
+     * Returns an appropriate output directory for the decompiled data based upon the provided {@link JavaClassFileModel}.
+     *
+     * This could be the same directory as the file itself, if the file is already in the output directory. If the .class file is referencing a file
+     * in the input directory, then this will be a classes folder underneath the output directory.
+     */
+    static File getOutputDirectoryForClass(GraphContext context, JavaClassFileModel fileModel)
```
Comment (mbriskar) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/decompiler/DecompilerUtil.java]: maybe I would state in the documentation that this is only the top-directory path (without package)

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/912/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/281/
Test PASSed.

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/decompiler/DecompilerUtil.java (hunk 2 above)
Comment (mbriskar) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/decompiler/DecompilerUtil.java]: done. thanks

**Pull request merged**